app/components/request/config-form.js

```diff
@@ -0,0 +1,69 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { task } from 'ember-concurrency';
+import { reads } from '@ember/object/computed';
+import BranchSearching from 'travis/mixins/branch-searching';
+import { bindKeyboardShortcuts, unbindKeyboardShortcuts } from 'ember-keyboard-shortcuts';
+
+const CONFIG = {
+  SOURCE: {
+    API: 'api'
+  },
+  JAVASCRIPT: 'javascript',
+  YAML: 'yaml',
+  JSON: 'json',
+};
+
+export default Component.extend(BranchSearching, {
+  classNames: ['request-configs-form'],
+
+  repo: reads('request.repo'),
+
+  keyboardShortcuts: {
+    'shift+enter': 'submit'
+  },
+
+  onChange() {},
+  onSubmit() {},
+
+  didInsertElement() {
+    this._super(...arguments);
+    bindKeyboardShortcuts(this);
+  },
+
+  willDestroyElement() {
+    this._super(...arguments);
+    unbindKeyboardShortcuts(this);
+  },
+
+  searchBranches: task(function* (query) {
+    const result = yield this.searchBranch.perform(this.get('repo.id'), query);
+    return result.mapBy('name');
+  }),
+
+  configMode: computed('config', function () {
+    const { config } = this;
+    const { JAVASCRIPT, YAML } = CONFIG;
+    return config && config.startsWith('{') ? JAVASCRIPT : YAML;
```

Comment:
I'm not sure I'm getting it right, but string that starts with `{` looks more like `JSON` to me rather than `JAVASCRIPT`. Just wanted to bring your attention here, to make sure the logic is correct.

Reply:
it seems the codemirror plugin requires the mode to be `javascript`, unfortunately, not `json` (there is no such file, and thus setting that mode fails ember's start up)

having said that, somewhere throughout all this refactoring branches and remerging/cherrypicking everything twice the setup for the editor modes was lost, and highlighting didn't work. i think i have fixed that now.

Reply:
Ah, I see, it makes sense then, thank you for explaining 👍